modules/drivers/oracle/src/metabase/driver/oracle.clj

```diff
@@ -99,31 +102,45 @@
   "The connection property used by the Oracle JDBC Thin Driver to control the program name."
   "v$session.program")
 
-(defn- handle-ssl-options [{:keys [ssl ssl-use-keystore ssl-use-truststore] :as details}]
+(defn- guess-keystore-type [^java.io.File keystore-file ^String password]
+  (try
+    (.getType (KeyStore/getInstance keystore-file (.toCharArray password)))
+    (catch Exception _
+      "JKS")))
```

Comment:
JKS is deprecated, this adds support for other keystore types like PKCS12.

Reply:
Could this impact people's existing configurations?

Reply:
We should probably catch `KeyStoreException` here to be more specific. Also maybe add some comments about why this is needed. I am able to open JKS keystores (generated with `keytool`) locally on openjdk 11.0.17. Is this just for certain JDKs?

Reply:
So far we have only supported `JKS` keystores. I'm assuming, that for existing working configurations this function returns `JKS`. I think it's extremely unlikely that people are successfully using keystores presented as `JKS` to the driver that are in realty of some other known type like `PKCS12` and the driver would break if we use `PKCS12` instead of `JKS`.

Still, if you know a better way to ensure compatibility or a way to test this hypothesis, please let me know.

Reply:
JKS keystores are deprecated (at least keytool warns against it if you create one), but supported. That's been the Java way so far. Since Java 9 the default format is PKCS12.

I'm using the fallback to JKS to be as sure as possible that the code is backwards compatible. What do you suggest we should do if we get an exception other than `KeyStoreException`?